src/events/tab_event_handler.ts

```diff
@@ -45,19 +47,35 @@ export class tabEventHandler {
 
 		// Get list of current tabs
 		const open_tabs = vscode.window.tabGroups.all;
-		const tabs: ITabFile[] = [] = open_tabs.flatMap(tab_group => 
+		const tabs: ITabFile[] = open_tabs.flatMap(tab_group => 
 			tab_group.tabs.map(tab => {
 				// Get path of tab
 				// @ts-ignore
-				const fileRelativePath = tab.input.uri.path;
+				const tabFilePath = tab.input.uri.path;
+				const splitPath = tabFilePath.split(`${this.repoPath}${path.sep}`);
+				if (splitPath.length == 1) {
+					CodeSyncLogger.info(`File from other repo detected, Skipping invalid tab: ${tabFilePath}`);
+					return null;
+				}
+				const tabValidator = new TabValidator();
+				if (!tabValidator.validateRepo(this.repoPath)) return;
```

Comment:
This should be the very first check, filePath split logic should come after this